pkg/hubble/metrics/api/api.go

```diff
@@ -95,18 +94,6 @@ func InitHandler(log logrus.FieldLogger, registry *prometheus.Registry, item *Na
 	return nil
 }
 
-// ProcessCiliumEndpointDeletion queries all handlers for a list of MetricVec and removes
```

Comment:
This instance of `ProcessCiliumEndpointDeletion` uses `DeleteMetricsAssociatedWithPod` but is only used in test code.
I'm curious where the code is that performs metric adjustment on pod deletion for product code.
It would seem here:
https://github.com/cilium/cilium/blob/62e26d498de76c734e8380ad9c1df7caf12a9d1a/pkg/hubble/metrics/metrics.go#L70-L75
but I can't see the analogous logic.

Reply:
You can look at 782f934641df5bafd4a9ee737e00872f65f56b64 to see how the code was originally implemented. Things have changed, removed, and renamed quite a bit since then. Previously, `hubblemetrics.ProcessPodDeletion(pod)` was called within `func (k *K8sWatcher) deleteK8sPodV1` in `pkg/k8s/watchers/pod.go` and `hubblemetrics.ProcessPodDeletion` called `ctx.DeleteMetricsAssociatedWithPod`. 

Now it seems that `ProcessCiliumEndpointDeletion` is called by `func (k *K8sCiliumEndpointsWatcher) endpointDeleted(endpoint *types.CiliumEndpoint)` from `pkg/k8s/watchers/cilium_endpoint.go` and the rest remains the same.

Reply:
Makes sense, thanks! Fixed in the new update.